src/nextjs/server/proxy.ts

```diff
@@ -49,13 +49,27 @@ export async function proxyAuthActionToConvex(
     `Fetching action ${action} with args ${JSON.stringify(args)}`,
     verbose,
   );
-  const untypedResult = await fetchAction(action, args, {
-    url: options?.convexUrl,
-    token,
-  });
 
   if (action === "auth:signIn") {
-    const result = untypedResult as SignInAction["_returnType"];
+    let result: SignInAction["_returnType"];
+    // Do not require auth when refreshing tokens or validating a code since they
+    // are steps in the auth flow.
+    const fetchActionAuthOptions =
```

Comment:
I added this just now to match how we use `authenticatedCall` and `unauthenticatedCall`. AFAICT the `signIn` function doesn't actually rely on being called as an authenticated call, but matching the existing behavior in the vanilla React integration seemed safer.